Lib/ensurepip/__init__.py

```diff
@@ -27,52 +17,36 @@
 _WHEEL_PKG_DIR = sysconfig.get_config_var('WHEEL_PKG_DIR')
 
 
-def _find_packages(path):
-    packages = {}
+def _find_wheel_pkg_dir_pip():
+    if _WHEEL_PKG_DIR is None:
+        return None
     try:
-        filenames = os.listdir(path)
+        filenames = os.listdir(_WHEEL_PKG_DIR)
     except OSError:
         # Ignore: path doesn't exist or permission error
-        filenames = ()
+        return None
     # Make the code deterministic if a directory contains multiple wheel files
     # of the same package, but don't attempt to implement correct version
     # comparison since this case should not happen.
-    filenames = sorted(filenames)
+    filenames = sorted(filenames, reverse=True)
```

Comment:
Why reverse? I guess it gives better odds of getting the correct version, but as the comment above says, we're not doing version comparison because it should never be needed.

Reply:
@AA-Turner ^

Reply:
I noted this in the commit message but not a PR comment; sorry. The implementation of this function now short-circuits and returns the first match, but that means we need to reverse the order as previously the final match would be returned, and we want to preserve that behaviour. We retain no guarantees about finding the correct version.